### ISSUE TITLE
SolaraViz docs

### ISSUE THREAD

**Issue by quaquel** [projectmesa/mesa#2494]
SolaraViz creates new model instances via `**model_parameters.value`. This implies that all inputs for instantiating a new model are keyword arguments. This is currently not well documented, and we might even consider adding this as an explicit check in `_check_model_params`.

**Comment by Corvince** [comment]
I agree that this should be documented and added to `_check_model_params`. Should be a good ticket for new contributors, so I tagged it. But I guess its not a high priority, since this has always been implicitly required (since mesa 0.x days), and as far as reported issues go never been an issue.

**Comment by Corvince** [comment]
Thats great to hear! Sorry for not responding earlier. If you have any specific question just ask, otherwise I am looking forward to your PR

**Comment by ashish-nagmoti** [comment]
@Corvince is final pr merged, ? (if it is then please close issue , as it's confusing)

**Comment by Spartan-71** [comment]
@quaquel one way to fix this is to add `*`(to catch all the positional parameters after self) in the model's `__init__` function. Once that is done then all the parameters after the `*` will be strictly keyword.

![Image](https://github.com/user-attachments/assets/a6bab996-df98-4f3b-a14c-b28ed86e45bd)

This seems to be a good approach to me, what are your thoughts?

**Comment by SiddharthBansal007** [comment]
Hey, is this issue still open. I would like to contribute.

**Comment by pragam-m25** [comment]
Hi, I'm new to Project Mesa and preparing for GSoC 2026. I see this issue is still open, though a few others have commented. If no one is actively working on a PR, I would love to take this up. I can start with the documentation update first.

**Comment by CODExSHAN29** [comment]
hey new to this project mesa and preparing for 2026 gsoc i saw this one is open want to contribute .

**Comment by pragam-m25** [comment]
Hi @CODExSHAN29, go ahead! I got occupied with other Mesa 3.0 explorations. Note  comment above—it seems to be an upstream issue with Solara, not just a simple doc fix. Good luck!

**Comment by Akshay-krish** [comment]
Hi, i would like to work on this issue if it is still available. I will review the current SolaraViz documentation and recommend necessary documentation updates

**Comment by Saranyadharani** [comment]
Hi, I’ve completed the implementation for this issue and opened a PR: #3165

- All tests for Solara visualization pass (`pytest tests/test_solara_viz.py`)
- Verified manually in VS Code
- Ready for review

This PR aligns with my GSoC 2026 proposal focusing on Solara visualization in Mesa.

PR link: https://github.com/mesa/mesa/pull/3165

**Comment by visha-loop** [comment]
Hi, i would like to work on the issue if its is still available .

**Comment by ApurvaPatil2401** [comment]
Hi! I've successfully set up my development environment in a GitHub Codespace and verified that the SolarViz tests are passing. I'd like to help improve the SolarViz documentation regarding the requirement for keyword model arguments. May I be assigned to this?

**Comment by deepanshu-maliyan** [comment]
Hi! I'd like to contribute to this issue. I plan to:
1. Update the SolaraViz documentation to explicitly mention that all model arguments must be keyword arguments (due to `**model_parameters.value`).
2. Add a check in `_check_model_params` to raise a clear error if positional-only parameters are detected in the model's `__init__`.

I'll follow the approach suggested by @Spartan-71 (adding `*` to catch positional parameters) and make sure the existing tests continue to pass. Let me know if there's anything specific you'd like me to focus on!

**Comment by Artemiz0307** [comment]
Hi, I’m interested in contributing to this issue as a first-time contributor.
Could I work on this?

**Comment by jaydipsharma** [comment]
Hi, I’d like to work on this issue if it’s still available. I’m making my first public open-source contribution and can prepare a docs update clarifying that SolaraViz creates model instances via `**model_parameters.value`, so model inputs need to be keyword arguments. Please let me know if there’s any preferred location or format for the docs change.
